tests/unit/test_sizer.cc

```diff
@@ -65,16 +65,27 @@ struct Test1 {
 TEST_F(TestSizer, test_sizer_1) {
```

Comment:
I don't think I see any tests that check for the failure case when the buffer size doesn't match. Could we construct that use case which would improve code coverage?